api/rest/bundle_handler.go

```diff
@@ -110,7 +131,38 @@ func (h BundleHandler) Create(w http.ResponseWriter, r *http.Request) {
 
 	//TODO(janisz): use context cancel function to cancel bundle creation https://jira.mesosphere.com/browse/DCOS_OSS-5222
 	ctx, _ := context.WithTimeout(context.Background(), h.bundleCreationTimeout) //nolint:govet
-	done := make(chan []string)
+
+	var args createArguments
+	if r.Body == nil {
+		// assume local
+		args.BundleType = bundleTypeLocal
+	} else {
+		err = json.NewDecoder(r.Body).Decode(&args)
+		if err != nil {
+			writeJSONError(w, http.StatusBadRequest, fmt.Errorf("could not decode JSON body: %s", err))
+			return
+		}
+	}
+
+	switch strings.ToLower(args.BundleType) {
+	case bundleTypeLocal:
+		done := make(chan []string)
+		h.createLocalBundle(ctx, &bundle, done, dataFile, stateFilePath)
+	case bundleTypeRemote:
+		err = h.createRemoteBundle(ctx, &bundle, args.Nodes, dataFile, stateFilePath)
+		if err != nil {
+			writeJSONError(w, http.StatusBadRequest, err)
```

Comment:
I'm not sure what you mean, if `createRemoteBundle` returns an error then the endpoint will return a 400 which you seemed to be agreeing with in your comment below. I think it makes some sense to inline it but regardless, I think the appropriate status code for all the errors it throws would be something in the 400 range.

Reply:
I agree this could be 400 error but it's hard to tell because you need to read underlying function to understand understand it. If we inline it it will be clear that 400 is reported because user posted no nodes.